teps/0094-configuring-resources-at-runtime.md

```diff
@@ -52,12 +61,472 @@ While this may be a valuable feature, it should be considered in a separate prop
 depending on the image or source being built.
 - Kubeflow pipelines and other data pipelines may have variable resource requirements
 depending on the data being processed.
-- Catalog Tasks should be generally reusable in different environments
+- Catalog `Task`s should be generally reusable in different environments
 that may have different resource constraints.
 
 ## Requirements
 
 - Users can specify `Step` and `Sidecar` resource requirements at runtime.
 - Users can specify `Step` and `Sidecar` resource requirements for `Task`s 
 or `Pipeline`s they don't own, especially those in the Catalog.
-- Users can specify resource requirements for individual `Step`s and `Sidecar`s.
\ No newline at end of file
+- Users can specify resource requirements for individual `Step`s and `Sidecar`s.
+
+## Proposal
+
+Augment `TaskRunSpec` and `PipelineTaskRunSpec` with a mapping of `Step` names to overrides
+and a mapping of `Sidecar` names to overrides.
+
+## Design Details
+
+```go
+import corev1 "k8s.io/api/core/v1"
+
+type TaskRunStepOverride struct {
+  // The name of the Step to override.
+  // If unspecified, applies to all Steps.
+  TaskRunStepName string
```

Comment:
The step name is not mandatory, so this implies that the resources of a step with no name set can only be set by setting resources for all steps. We'll have to document this limitation.
I wonder if we should give users and option to refer to a step by index too?

Reply:
Heh, I see there is a comment about this below.